### PR TITLE
Add _check_option function

### PR DESCRIPTION
**Pull request by wmvanvliet**
As @massich noted in #5946, the following snippet is a common occurrence in our codebase:

```python
# Validate method parameter
if method not in ['foo', 'bar', 'baz']:
    raise ValueError("Invalid value for 'method' parameter. Valid options are ...")
```

Generating a good error message takes a bit more work. You can either do:
```python
valid_options = ['foo', 'bar', 'baz']
if method not in valid_options:
    raise ValueError("Invalid value for 'method' parameter. Valid options are %s, "
                     "but got '%s' instead." % (valid_options, method))
```

or go just write it all out if you don't like the list-like formatting of the valid options:

```python
if method not in ['foo', 'bar', 'baz']:
    raise ValueError("Invalid value for 'method' parameter. Valid options are "
                     "'foo', 'bar' and 'baz', but got '%s' instead." % method)
```

It's a little bit of a hassle and we're not very consistent across the code base in how we generate such errors.

This PR adds a private `_check_option` function that you can use to perform such a check and generate a friendly error message in a single line:

```python
_check_option('method', method, valid_options=['foo', 'bar', 'baz'])
```

Small QOL improvement or unnecessary bloat? You be the judge :)
If we agree that we want this function, I'll open a follow-up PR that changes the bulk of the occurrences of this pattern in our codebase.

### CODE REVIEW

**Comment by larsoner** [comment]
+1 from me

File: mne/utils/check.py [hunk 1]
```diff
@@ -432,3 +432,44 @@ def _check_one_ch_type(info, picks, noise_cov, method):
     elif method == 'dics' and sum(ch_types) > 1:
         warn('The use of several sensor types with the DICS beamformer is '
              'not heavily tested yet.')
+
+
+def _check_option(parameter, value, allowed_values, error=ValueError):
+    """Checks a the value of a parameter against a list of valid options.
```
Comment (agramfort) [inline mne/utils/check.py]: a the -> the

**Comment by agramfort** [comment]
can you use/integrate this code anywhere? Let's clean at the same time.

**Comment by massich** [comment]
+1000

I found myself refactoring again here https://github.com/mne-tools/mne-python/pull/5836/files#diff-d8a578281e14fb43fe3a386a41036b8fR794

So if you had not opened this PR I would have done it.

I would put the word `rise` in the function name and maybe we could add the type of error to raise. But if you have a bulk of examples, maybe we can just start by this and then check if we can refactor further.

File: mne/utils/tests/test_check.py [hunk 2]
```diff
@@ -102,3 +102,32 @@ def test_check_info_inv():
     picks = _check_info_inv(epochs.info, forward, noise_cov=noise_cov,
                             data_cov=data_cov)
     assert list(range(7, 10)) == picks
+
+
+def test_check_option():
```
Comment (massich) [inline mne/utils/tests/test_check.py]: This is a good usecase to use pytest.mark.parametrize

Comment (wmvanvliet) [reply mne/utils/tests/test_check.py]: Thanks for pointing me to this function. However, using `pytest.mark.parametrize` ended up not being any shorter nor clearer than these four lines. A little too much changes across each line.

Comment (wmvanvliet) [reply mne/utils/tests/test_check.py]: ...or I just don't understand the `parametrize` function too well. If you have some clever refactoring in mind, please share :)

**Comment by wmvanvliet** [comment]
> I would put the word `rise` in the function name

(I think you meant `raise`)

I totally see how that is good practice. But in this case, all the functions in the `utils.check` module start with `_check_` (and many raise errors). Hence I feel that in this particular case, consistency beats  this otherwise excellent guideline.

**Comment by wmvanvliet** [comment]
> can you use/integrate this code anywhere? Let's clean at the same time.

I'll put in a few use cases in this PR. But shall we leave be big cleanup for a separate PR that will strictly be a search and replace?

**Comment by agramfort** [comment]
git grep "Got "

should help find some places you can use this new check function

I would favor to do the clean up in this PR. But others can help by pushing to the PR directly.

Just saying :)

**Comment by wmvanvliet** [comment]
> I would favor to do the clean up in this PR.

Then you got it!

**Comment by agramfort** [comment]
then I go for it or you go for it?

**Comment by wmvanvliet** [comment]
Lets both go for it! There are plenty of cases to cover I guess :) And I'm still on paternity leave so my working time is spotty.

**Comment by massich** [comment]
Looking at the API I would prefer something like this:
```py
def _check_option(*, option, accepted_values, error):
       raise ...

my_var = None
_check_option(option=my_var, accepted_values=[Ture, False], error=ValueError) 
```
and that it prints `"valid values for my_var are "True" or "False"; got None"`
I comparing it to `_check_option('my_var', my_var, [Ture, False])` I find that if someone knows nothing about the whatever does `_check_option` reads the first one it is clear what it does (and that it will raise an error).

(We can allow for not passing option name `_check_option(my_var, accepted_values=[Ture, False], error=ValueError)`. Also I'm not suggesting that we should put `*` to force it. This is just to illustrate what I'm refering to and then the rest be a convetion. We can even inforce such convention with LGTM)

**Comment by larsoner** [comment]
I prefer it the way it is currently. It's really an expert/dev option and I don't find all the extra chars in our repo will be helpful in the long run. Extra verbosity can hurt readability sometimes, it's a trade-off. Here I think the first is better for something that is meant to be a time-saver anyway.

File: mne/utils/check.py [hunk 3]
```diff
@@ -432,3 +432,40 @@ def _check_one_ch_type(info, picks, noise_cov, method):
...
+
+    Raises an exception with a readable error message if the value was invalid.
+
+    Parameters
+    ----------
+    parameter : str
+        The name of the parameter to check. This is used in the error message.
+    value : str
+        The value of the parameter to check.
+    allowed_values : list of str
```
Comment (massich) [inline mne/utils/check.py]: list ~of str~

File: mne/utils/check.py [hunk 4]
```diff
@@ -432,3 +432,40 @@ def _check_one_ch_type(info, picks, noise_cov, method):
...
+        The list of allowed values for the parameter.
+    error : type of Exception
+        The error type to raise if the value was invalid. Defaults to
+        ``ValueError``.
+
+    Returns
+    -------
+    is_valid : bool
+        Whether the value was valid or not.
```
Comment (massich) [inline mne/utils/check.py]: if its not valid, you are not returning.

Comment (wmvanvliet) [reply mne/utils/check.py]: Good point.

**Comment by agramfort** [comment]
API of private function is fine with me as is. Let's use it as much as possible everywhere now.

**Comment by wmvanvliet** [comment]
That's filenames stating with A-E done. Lots more to do.

File: mne/channels/layout.py [hunk 5]
```diff
@@ -382,10 +382,7 @@ def find_layout(info, ch_type=None, exclude='bads'):
     layout : Layout instance | None
         None if layout not found.
     """
-    our_types = ' or '.join(['`None`', '`mag`', '`grad`', '`meg`'])
-    if ch_type not in (None, 'meg', 'mag', 'grad', 'eeg'):
-        raise ValueError('Invalid channel type (%s) requested '
-                         '`ch_type` must be %s' % (ch_type, our_types))
+    _check_option('ch_type', ch_type, [None, 'mag', 'grad', 'meg'])
```
Comment (agramfort) [inline mne/channels/layout.py]: 'eeg' missing

Comment (wmvanvliet) [reply mne/channels/layout.py]: good catch!

**Comment by larsoner** [comment]
This pull request **introduces 1 alert** when merging 692fca10a017085074a931db543e92dc4d55f94f into 541860518239b541eaf7be85db271327ec90085c - [view on LGTM.com](https://lgtm.com/projects/g/mne-tools/mne-python/rev/pr-25ee335e160293cd8ed4bc1820208968a4197d55)

**new alerts:**

* 1 for Wrong number of arguments in a call

---

*Comment posted by [LGTM.com](https://lgtm.com)*

File: mne/utils/check.py [hunk 6]
```diff
@@ -461,12 +458,13 @@ def _check_option(parameter, value, allowed_values, error=ValueError):
...
+        options = 'The only allowed value is %r' % allowed_values[0]
     else:
-        options = "Allowed values are "
-        options += ", ".join(["'%s'" % v for v in allowed_values[:-1]])
-        options += " and '%s'" % allowed_values[-1]
-    raise error(msg.format(parameter=parameter, options=options, value=value))
+        options = 'Allowed values are '
+        options += ', '.join(['%r' % v for v in allowed_values[:-1]])
+        options += ' and %r' % allowed_values[-1]
+    raise ValueError(msg.format(parameter=parameter, options=options,
```
Comment (massich) [inline mne/utils/check.py]: are you sure that it will always be a `ValueError`? I'm fine with the change and change it back if a TypeError appears.

Comment (massich) [reply mne/utils/check.py]: Scratch that. Actually to check if an option is a subclass of the `allowed_values` the function should be quite different: 

```py
def _check_option(option_name, option_val, allowed_values, type_of_check='one of', ...):
```

and allow for `type_of_check='sublcass'`, etc..

Comment (wmvanvliet) [reply mne/utils/check.py]: Why would I want to check whether an option is a subclass of `allowed_values`?

I've gone through the entire codebase now. This function only has to deal with `str`, `None`, `True` and `False`. It also only needed to throw `ValueError`, hence I removed the code for throwing other types of errors.

The reason this function doesn't need to be more complicated than this, is that all it really does is form a very specific error message that just happens to be a very common one. When a different kind of check is required, the error message is likely to be different as well and you're better off not bending over just to use this function but just write the if-statement and `throw` yourself.

Comment (agramfort) [reply mne/utils/check.py]: I agree. Let’s not be too smart here. @wmvanvliet can you just see why CIs complain ?

File: mne/utils/check.py (hunk 4 above)
Comment (agramfort) [reply mne/utils/check.py]: I would remove the "Returns" section as it returns None or raises a ValueError.

besides LGTM

**Comment by wmvanvliet** [comment]
I think I'm done here. Ready for merge!

File: mne/utils/check.py (hunk 4 above)
Comment (agramfort) [reply mne/utils/check.py]: @wmvanvliet this last one please

**Comment by larsoner** [comment]
Thanks @wmvanvliet !

**Comment by massich** [comment]
THX @wmvanvliet

**Pull request merged**